xaitk_saliency/impls/perturb_image/rise.py

```diff
@@ -0,0 +1,90 @@
+from xaitk_saliency.interfaces.perturb_image import PerturbImage
+import PIL.Image
+from typing import Optional, Tuple, List, Dict, Any
+import numpy as np
+from skimage.transform import resize
+
+
+class RISEPertubation (PerturbImage):
+    def _generate_masks(self, input_size: Tuple[int, int]) -> np.ndarray:
+        """
+        Randomly crop, upscale, and interpolate binary masks to generate a set
+        of random masks to apply to the image.
+        :param (int, int) input_size:
+            Size of the model's input as (rows, cols). Binary masks are upsampled to
+            this resolution to be applied to (multiplied with) the input later.
+            E.g. (224, 224).
+        """
+        masks = np.empty((self.N, *input_size))
+        cell_size = np.ceil(np.array(input_size) / self.s)
+        # Upscale factor
+        up_size = (self.s + 1) * cell_size
+        for i in range(self.N):
+            # Random shifts
+            x = self.rng.integers(0, cell_size[0])
+            y = self.rng.integers(0, cell_size[1])
+            # Linear upsampling and random cropping
+            masks[i, :, :] = resize(self.grid[i], up_size, order=1, mode='reflect',
+                                    anti_aliasing=False)[x:x + input_size[0], y:y + input_size[1]]
+        # Reshape brings this to: (N, input_size[0], input_size[1], 1)
+        return masks
+
+    def __init__(
+        self,
+        N: int,
+        s: int,
+        p1: float,
+        seed: Optional[float] = None
+    ):
+        """
+        Generate a set of random binary masks
+        :param int N:
+            Number of random masks used in the algorithm. E.g. 1000.
+        :param int s:
+            Spatial resolution of the small masking grid. E.g. 8.
```

Comment:
Might be good to note that this is a "side length" and the resulting grid is square.